config/v1/types_infrastructure.go

```diff
@@ -1417,6 +1419,89 @@ type NutanixPlatformSpec struct {
 	// +listType=map
 	// +listMapKey=name
 	PrismElements []NutanixPrismElementEndpoint `json:"prismElements"`
+
+	// failureDomains configures failure domains information for the Nutanix platform.
+	// When set, the failure domains defined here may be used to spread Machines across
+	// prism element clusters to improve fault tolerance of the cluster.
+	// +listType=map
+	// +listMapKey=name
+	// +optional
+	FailureDomains []NutanixFailureDomain `json:"failureDomains"`
+}
+
+// NutanixFailureDomain configures failure domain information for the Nutanix platform.
+type NutanixFailureDomain struct {
+	// name defines the unique name of a failure domain.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=64
+	// +kubebuilder:validation:Pattern=`[a-z0-9]([-a-z0-9]*[a-z0-9])?`
+	Name string `json:"name"`
+
+	// prismCentralEndpoint holds the endpoint (address, port) to access the Nutanix Prism Central.
+	// If this is not configured, it means this failure domain uses the default prism_central
+	// endpoint as configured in the nutanix platformSpec.
```

Comment:
Do you expect, when multiple failure domains are configured, that at most one of the failure domains should have this field omitted? I'm wondering if we should just make it required if that's the expectation. If we expect there's a valid case where all failure domains use the same prism central then we can continue with it as optional

Reply:
For OCP 4.15, we only support a single PC with multi-PE. The prismCentralEndpoint here in the FailureDomain is just a placeholder for the future multi-PC support, so it should be optional. We will document this in the 4.15 release notes and documentation.

Reply:
So is that to say, it will always be optional? Multi-pc will be some users case and some others will just have the one?

If we only support a single one today, I think it may be better to drop this field out until we support the multi-pc case

Reply:
Dropped this for now.